lnd_test.go

```diff
@@ -9092,6 +9089,20 @@ func testMultiHopReceiverChainClaim(net *lntest.NetworkHarness, t *harnessTest)
 		t.Fatalf(predErr.Error())
 	}
 
+	// The invoice should show as settled for Carol, even if it was swept
+	// on-chain.
+	invoicesReq := &lnrpc.ListInvoiceRequest{}
+	invoicesResp, err := carol.ListInvoices(ctxb, invoicesReq)
+	if err != nil {
+		t.Fatalf("unable to retrieve invoices: %v", err)
+	}
+	if len(invoicesResp.Invoices) != 1 {
+		t.Fatalf("expected 1 invoice, got %d", len(invoicesResp.Invoices))
+	}
+	if !invoicesResp.Invoices[0].Settled {
+		t.Fatalf("expected invoice to be settled")
+	}
+
```

Comment:
You could consider adding test coverage on the unit level. In general, I myself prefer to test things only in an integration test if there is really no other option. 

I have an (unmerged) contract resolver test for the outgoingcontestresolver, https://github.com/lightningnetwork/lnd/pull/1875/commits/33c348b369143d07132d3b015478f0811d24deed. 

But I can also imagine that you consider this too much effort in comparison to the size of the change.

Reply:
I opted for the integration test change as we're interested in making sure that the invoice state is reflected correctly within the database. I guess we could have this done at the unit test level though. Let me know if you'd like to proceed with that instead.